worldedit-core/src/main/java/com/sk89q/worldedit/function/operation/ForwardExtentCopy.java

```diff
@@ -267,10 +265,13 @@ public Operation resume(RunContext run) throws WorldEditException {
                     }
                 }
                 EntityVisitor entityVisitor = new EntityVisitor(entities.iterator(), entityCopy);
-                return new DelegateOperation(this, new OperationQueue(blockVisitor, entityVisitor));
-            } else {
-                return new DelegateOperation(this, blockVisitor);
+                operation = new DelegateOperation(this, new OperationQueue(blockVisitor, entityVisitor));
```

Comment:
Has this change been fully tested? Feels weird to remove creation of the blockVisitor delegate

Reply:
Hi. I did not remove creation of the blockVisitor delegate.
I've just moved lines:
`lastVisitor = blockVisitor;`
`currentTransform = currentTransform.combine(transform);`
to the end (after "copyingEntities" block)

I've tested it. Entities are now rotated perfectly. But I do not know was it "fully" test or not.
Problem in original code was that line:
'currentTransform = currentTransform.combine(transform);'
was before copying entities:
'ExtentEntityCopy entityCopy = new ExtentEntityCopy(from, destination, to, currentTransform);'

So we need to recalculate `currentTransform` after copying entities.
I moved `currentTransform` after copying entities.
`lastVisitor` initialization was moved with `currentTransform` optionaly as they were together in orogonal code :)

My actioans step by step:

Original code. Problem: `currentTransform` is changed before `new ExtentEntityCopy`
```
            lastVisitor = blockVisitor;
            currentTransform = currentTransform.combine(transform);

            if (copyingEntities) {
                ExtentEntityCopy entityCopy = new ExtentEntityCopy(from, destination, to, currentTransform);
                entityCopy.setRemoving(removingEntities);
                List<? extends Entity> entities = source.getEntities(region);
                // Switch to entities.removeIf after Java 8 cutoff.
                Iterator<? extends Entity> entityIterator = entities.iterator();
                while (entityIterator.hasNext()) {
                    EntityType type = entityIterator.next().getFacet(EntityType.class);

                    if (type != null && !type.isPasteable()) {
                        entityIterator.remove();
                    }
                }
                EntityVisitor entityVisitor = new EntityVisitor(entities.iterator(), entityCopy);
                return new DelegateOperation(this, new OperationQueue(blockVisitor, entityVisitor));
            } else {
                return new DelegateOperation(this, blockVisitor);
            }
```
First step (preparation. **NO LOGIC CHANGE!**). Taking both 'return' statements outside `if-else` block:
```
            lastVisitor = blockVisitor;
            currentTransform = currentTransform.combine(transform);

>>>>>>      Operation operation;
            if (copyingEntities) {
                ExtentEntityCopy entityCopy = new ExtentEntityCopy(from, destination, to, currentTransform);
                entityCopy.setRemoving(removingEntities);
                List<? extends Entity> entities = source.getEntities(region);
                // Switch to entities.removeIf after Java 8 cutoff.
                Iterator<? extends Entity> entityIterator = entities.iterator();
                while (entityIterator.hasNext()) {
                    EntityType type = entityIterator.next().getFacet(EntityType.class);

                    if (type != null && !type.isPasteable()) {
                        entityIterator.remove();
                    }
                }
                EntityVisitor entityVisitor = new EntityVisitor(entities.iterator(), entityCopy);
>>>>>>          operation = new OperationQueue(blockVisitor, entityVisitor);
            } else {
>>>>>>          operation = blockVisitor;
            }
>>>>>>      return new DelegateOperation(this, operation);
```
Second step (preparation. **NO LOGIC CHANGE!**). Removing `else` block:
```
            lastVisitor = blockVisitor;
            currentTransform = currentTransform.combine(transform);

>>>>>>      Operation operation = blockVisitor;
            if (copyingEntities) {
                ExtentEntityCopy entityCopy = new ExtentEntityCopy(from, destination, to, currentTransform);
                entityCopy.setRemoving(removingEntities);
                List<? extends Entity> entities = source.getEntities(region);
                // Switch to entities.removeIf after Java 8 cutoff.
                Iterator<? extends Entity> entityIterator = entities.iterator();
                while (entityIterator.hasNext()) {
                    EntityType type = entityIterator.next().getFacet(EntityType.class);

                    if (type != null && !type.isPasteable()) {
                        entityIterator.remove();
                    }
                }
                EntityVisitor entityVisitor = new EntityVisitor(entities.iterator(), entityCopy);
                operation = new OperationQueue(blockVisitor, entityVisitor);
>>>>>>      }
            return new DelegateOperation(this, operation);
```
Final step. Moving `lastVisitor` and `currentTransform` recalculation after `if`-block.
```
            Operation operation = blockVisitor;
            if (copyingEntities) {
                ExtentEntityCopy entityCopy = new ExtentEntityCopy(from, destination, to, currentTransform);
                entityCopy.setRemoving(removingEntities);
                List<? extends Entity> entities = source.getEntities(region);
                // Switch to entities.removeIf after Java 8 cutoff.
                Iterator<? extends Entity> entityIterator = entities.iterator();
                while (entityIterator.hasNext()) {
                    EntityType type = entityIterator.next().getFacet(EntityType.class);

                    if (type != null && !type.isPasteable()) {
                        entityIterator.remove();
                    }
                }
                EntityVisitor entityVisitor = new EntityVisitor(entities.iterator(), entityCopy);
                operation = new OperationQueue(blockVisitor, entityVisitor);
            }

>>>>>>      lastVisitor = blockVisitor;
>>>>>>      currentTransform = currentTransform.combine(transform);

            return new DelegateOperation(this, operation);
```
Moving `currentTransform` - **LOGIC CHANGE!** Prevent "double" transformation before `new ExtentEntityCopy`
Moving `lastVisitor` - **NO LOGIC CHANGE!** This is optionaly. Just to leave `lastVisitor` and `currentTransform` together as it was in orgonal code.

So `lastVisitor` and `currentTransform` has the same values as they have them in original code.
I did it on purpose, because I do not understand the point in maintaining the double transformation (`currentTransform`) after the return 

Sorry for english :)